interface/admin.py

```diff
@@ -1,6 +1,44 @@
+from tarfile import TarFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
 from django.contrib import admin
+from django.http import HttpResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, _assignment):
+        assignment = _assignment[0]
+        ta_user = request.user
+        ta_group = ta_user.groups.all()[0].name
+
+        submissions = Submission.objects.filter(
+                            assignment=assignment,
+                            user__groups__name=ta_group).exclude(user=ta_user)
+
+        with TemporaryDirectory() as _tmp:
+            tmp = Path(_tmp)
+
+            with TarFile(tmp / 'review.tar', 'x') as tar:
```

Comment:
Why not just provide the zip instead?

Reply:
Use a `zip` archive instead of the `tar` archive? Wasn't the zip in a zip idea bad?

Reply:
> Wasn't the zip in a zip idea bad?

That was in another context, when you were creating a payload with a payload inside.

> Why not just provide the zip instead?

I think he's adding a bunch of zips, one for each submission? This is basically bulk download.

But for this use case, I'd go with zip, it's easier to work with for the non-geek user.